packages/dashboard/src/components/composed/RPCs/RPC/Details/Expanded/DecodedParams.tsx

```diff
@@ -0,0 +1,52 @@
+import { Group, Text, createStyles } from "@mantine/core";
+import { Prism } from "@mantine/prism";
+import { AlertTriangle } from "react-feather";
+import type { CalldataDecoding } from "@truffle/codec";
+import { inspectDecoding } from "src/utils/dash";
+import type { DecodableRpcMethod } from "src/utils/constants";
+
+const useStyles = createStyles((theme, _params, _getRef) => ({
+  alert: {
+    color: theme.colors.orange[6]
+  }
+}));
+
+type DecodedParamsProps = {
+  decoding: CalldataDecoding | string;
+  decodingSucceeded: boolean;
+  method: DecodableRpcMethod;
+};
+
+function DecodedParams({
+  decoding,
+  decodingSucceeded,
+  method
+}: DecodedParamsProps): JSX.Element {
+  const decodingInspected = inspectDecoding(decoding);
+  const { classes } = useStyles();
+
+  const language =
+    method === "eth_sendTransaction"
+      ? "javascript"
+      : method === "eth_signTypedData_v3" || method === "eth_signTypedData_v4"
+      ? "json"
+      : "markdown";
```

Comment:
I guess maybe this should at least be its own function (e.g. `highlightLanguageForRpcMethod`)

Reply:
Makes sense. [Updated](https://github.com/trufflesuite/truffle/commit/ecc8ea83001946da7e2398853d8941425e75f101)